addons/website/static/src/snippets/s_rate/options.js

```diff
@@ -0,0 +1,41 @@
+
+odoo.define('website.s_rate_options', function (require) {
+'use strict';
+
+var options = require('web_editor.snippets.options');
+
+options.registry.rate = options.Class.extend({
+    /**
+     * @override
+     */
+    start: function () {
+        console.log('ok');
+        const descriptionHtml = '<p class="s_rate_dish_description o_default_snippet_text d-none">Add here a dish description</p>';
+        if (this.$target.find('.s_rate_dish_description').length === 0) {
+            this.$target.find('.s_rate_content').after(descriptionHtml);
+        }
+        return this._super.apply(this, arguments);
+    },
+    /**
+     * @override
+     */
+    destroy: function () {
+        this._super.apply(this, arguments);
+    },
+    /**
+     * @override
+     */
+    onBuilt: function () {
+        this.$target.find('.s_rate_dish_description').removeClass('d-none');
+    },
+    /**
+     * @override
+     */
+    cleanForSave: function () {
+        let $descriptions = this.$target.find('.s_rate_dish_description');
+        if ($descriptions.hasClass('d-none') && $descriptions.hasClass('o_default_snippet_text')) {
+            $descriptions.remove();
+        }
```

Comment:
In fact, if someone edit half the description and then choose hide description, he may not want to lose them all. Only remove the one that are hidden and using a default snippet text.

In fact saying this, I think we should do this: no more start, no more on build, no cleanForSave, only a new method option for your snippet like "toggleDescription" if show -> add all missing descriptions and show existing ones, if hide -> remove all descriptions with default text and hide the ones with edited text. And then implement the `_computeWidgetState`.